src/BlockchainData.js

```diff
@@ -0,0 +1,228 @@
+import React, { Component } from 'react';
+import Web3 from 'web3';
+import Tx from 'ethereumjs-tx';
+import EthCrypto from 'eth-crypto';
+import { MULTISIG_ABI, MULTISIG_ADDRESS, MULTISIG_ERC20_ABI, MULTISIG_ERC20_ADDRESS, ERC20_ABI, 
+	TOKENIMPERIAL_ADDRESS, TOKENDEMOCRATIC_ADDRESS, TOKENIMPERIAL_SYMBOL, TOKENDEMOCRATIC_SYMBOL } from './config';
+import SmartContract from './SmartContract'; 
+import ERC20 from './erc20/ERC20';
+import './css/ButtonGroup.css';
+
+
+/* Validate forms, clear data in forms, check that transaction transfer works correctly*/ 
+class BlockchainData extends Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			web3: '',
+			multisig: '',
+			etherAddress: '',
+			etherBalance: '',
+			tokenImperialBalance: '',
+			tokenDemocraticBalance: '',
+			contractBalance: '',
+			tokenImperialContractBalance: '',
+			tokenDemocraticContractBalance: '',
+			Ethereum: true
+		}
+	}
+	static getDerivedStateFromProps(props, state) {
+    	return {etherAddress: props.etherAddress };
+  	}
+
+  	componentDidMount() {
+  		const web3 = new Web3('https://ropsten.infura.io/v3/a33baa265ae340c29c82373e91533edf');
+  		this.setState({ web3: web3 });
+  		const multisig = new web3.eth.Contract(MULTISIG_ABI, MULTISIG_ADDRESS);
+		this.setState({ multisig: multisig });
+		const multisigERC20Token = new web3.eth.Contract(MULTISIG_ERC20_ABI, MULTISIG_ERC20_ADDRESS);
+		this.setState({ multisigERC20Token: multisigERC20Token });
+		const tokenImperial = new web3.eth.Contract(ERC20_ABI, TOKENIMPERIAL_ADDRESS);
+		this.setState({ tokenImperial: tokenImperial });
+		const tokenDemocratic = new web3.eth.Contract(ERC20_ABI, TOKENDEMOCRATIC_ADDRESS);
+		this.setState({ tokenDemocratic: tokenDemocratic });
+		web3.eth.getBalance(this.state.etherAddress, (error, balance) => {
+			if (error) {
+				console.log(error)
+			} else {
+		        this.setState({ etherBalance: balance });
+	    		console.log('Account ether balance: ', this.state.etherBalance);
+	    	}
+			})
+		multisigERC20Token.methods.tokenBalances(this.state.etherAddress, TOKENIMPERIAL_SYMBOL).call({from: this.state.etherAddress}, (error, balance) => {
+			if (error) {
+		        console.log(error)
+		    } else {
+		        this.setState({ tokenImperialContractBalance: balance });
+		       	console.log('The imperial token contract balance of account is: ', this.state.tokenImperialContractBalance);
+		    }
+		})
+
+		multisigERC20Token.methods.tokenBalances(this.state.etherAddress, TOKENDEMOCRATIC_SYMBOL).call({from: this.state.etherAddress}, (error, balance) => {
+			if (error) {
```

Comment:
you have a lot of pretty same blocks of code like.
Think about create function with few parameters and just call it 

```
const setBalance = (balance, error, message) => {
if (error) {
		        console.log(error)
		    } else {
		        this.setState({ tokenImperialContractBalance: balance });
		       	console.log(message, this.state.tokenImperialContractBalance);
		    }
}

```